File: commands/ping.js
const Command = require('../classes/Command');

module.exports = class ping extends Command {
    constructor(client) {
        super(client, {
            name: 'ping',
            description: 'Obtenir le Ping du bot',
            aliases: [
                'pong'
            ]
        });
    }
    async run(message) {
        let send = Date.now();
        message.channel.send('Mesure...').then(m => {
            m.edit(`Pong! En ${Date.now() - send}ms`);
        })
    }
};

File: src/commands/ping.rs
use serenity::client::Context;
use serenity::framework::standard::{macros::command, CommandResult};
use serenity::model::channel::Message;
use serenity::utils::Colour;
use std::time::Instant;

#[command]
async fn ping(ctx: &Context, msg: &Message) -> CommandResult {
    let dur = Instant::now();
    let mut botmsg = msg.reply(ctx, "Calcul...").await?;
    botmsg
        .edit(ctx, |m| {
            m.content("Pong!");
            m.embed(|e| {
                e.title("Ping du bot: ");
                e.description(format!(
                    "{} ms ({} µs)",
                    dur.elapsed().as_millis(),
                    dur.elapsed().as_micros()
                ));
                e.colour(Colour::DARK_GREEN);
                e
            })
        })
        .await?;
    Ok(())
}

